### ISSUE TITLE
Improve Webpack workflow

### ISSUE THREAD

**Issue by iamakulov** [ilyabirman/Likely#43]
Currently, Webpack users have to install `json-loader` to use Likely in their projects (see [“Using with Webpack”](https://github.com/ilyabirman/Likely#using-with-webpack)). This is required because `main` in `package.json` points to `source/index.js`, and one of the source files `require`s a JSON file. The requirement to install `json-loader` is unfriendly, and we want to remove it.

To fix this, we should switch to UMD builds and make `main` point to `release/likely.js`. Currently, we use Browserify for the build process. I’ve googled a bit, and it seems there’s no easy way to create a UMD build with Browserify. So most likely we’ll need to switch to Webpack.

**PRs are welcome.** Feel free not to switch to Webpack if you find an appropriate way to create a UMD build with Browserify.

**The PR should also update the “Using with Webpack” section in the Readme.** As #40 that updates the Readme is still open, you are probably better to rebase upon [the `update-readme` branch](https://github.com/ilyabirman/Likely/tree/update-readme).

**Comment by vitkarpov** [comment]
> one of the source files requires a JSON file

It's an `svg.json` file with icons hash map. Maybe this should be a regular commonjs-module?

```js
module.exports = {
  ''facebook": "....",
  "twitter": "...",
  ....
}
```

Seems in this case webpack doesn't need any custom loaders, right?

**Comment by iamakulov** [comment]
Yup, seems like an appropriate solution. Feel free to submit a PR if you’d like to :-)

**Comment by iamakulov** [comment]
Wait, nope.

**Comment by vitkarpov** [comment]
What? :)

**Comment by vitkarpov** [comment]
Btw, if we have a webpack anyway, why we need a browserify build? Maybe we need to switch to webpack anyway?

**Comment by iamakulov** [comment]
> Wait, nope.

I’ve thought that we’d still want to implement a UMD build in the future, but in fact there’s no need to do this until we have a real request. So, if you think you have time to switch Likely to Webpack, that’d be great because this’ll make us more flexible in the future. If you don’t, a simple `svg.json` → `svg.js` solution is OK.

**Comment by vitkarpov** [comment]
I think that there're two different problems:

- remove json to simplify current webpack flow
- get rid of browserify and keep use only webpack

If it's right, first of all, seems right to get rid of json file and then make a new issue about switching to Webpack.

**Comment by iamakulov** [comment]
> Btw, if we have a webpack, why we need a browserify build?

Important: we don’t use webpack at the moment. Only Browserify.

> If it's right, first of all, seems right to get rid of json file and then make a new issue about switching to webpack.

Yep, you’re right. I’ll create an issue about migration to Webpack.

**Comment by iamakulov** [comment]
@vitkarpov, if you take this issue, please write back here so nobody else also takes it by an accident.

**Comment by vitkarpov** [comment]
Yep. I'll take it. Seems it's time to add me to the repo, so I can be assigned to issues :)

**Comment by iamakulov** [comment]
> Seems it's time to add me to the repo, so I can be assigned to issues :)

I’ll ask Ilya about this :-)

**Comment by vitkarpov** [comment]
I think we should merge `update-reame` first and then I'll update the webpack section.

**Comment by iamakulov** [comment]
OK, I’ll let #43 hang out then until we merge #40.

**Comment by vitkarpov** [comment]
Okay. I'll make additional changes in #43 after that.

**Comment by iamakulov** [comment]
@vitkarpov, you’re now a collaborator! I’ve assigned you to this issue.

**Comment by vitkarpov** [comment]
Thanks :) I can fix the readme now, #40 has been merged.

**Comment by vitkarpov** [comment]
Done
